### PR TITLE
exclude swift databag

### PR DESCRIPTION
**Pull request by MaximilianMeister**
any idea where this databag comes from?

### CODE REVIEW

**Comment by mjura** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Comment by vuntz** [comment]
+1

**Comment by MaximilianMeister** [comment]
rebased

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 1]
```diff
@@ -285,8 +285,9 @@ def restore_database
       end
 
       def proposal?(file)
-        !file.basename.to_s.match(/(_network\.json$)|(^template-(.*).json$)|(^queue\.json$)/) && \
-          file.to_s =~ /knife\/databags\/crowbar/
+        !file.basename.to_s.match(
+          /(_network\.json$)|(^template-(.*).json$)|(^queue\.json$)|(^swift\.json$)/
+        ) && file.to_s =~ /knife\/databags\/crowbar/
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: Coming back to this after some more thoughts... I don't have a Cloud 5 data dump here, so I cannot check, but I think all proposals are actually starting with `bc-`. I feel that would be the right check, actually.

Is there a data dump somewhere to look at for this?

Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: In the dump they still have the `bc-` prefix, that's correct. but in `upgrade.rb` which runs before restore, they are renamed. https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/upgrade.rb#L96

Comment (rhafer) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: How about actually loading the file and check if it has an "attributes" and a "deployment" section? That way we can be pretty sure it's a proposal.

**Comment by vuntz** [comment]
Okay, +1 for now and happy to get an improvement in a followup PR (or here if you have time, of course :-)).

**Comment by MaximilianMeister** [comment]
> How about actually loading the file and check if it has an "attributes" and a "deployment" section? That way we can be pretty sure it's a proposal

@rhafer @vuntz could you please check the last commit? if it is better like that I can squash it. I personally like the idea by @rhafer

**Comment by vuntz** [comment]
@MaximilianMeister looks good on the paper, +1 if you tested it :-)

**Comment by MaximilianMeister** [comment]
> @MaximilianMeister looks good on paper, +1 if you tested it :-)

sorry, was still not quite correct. now it should be fine

**Comment by MaximilianMeister** [comment]
actually not.. as we don't filter out templates anymore

**Comment by MaximilianMeister** [comment]
@rhafer @vuntz not sure what is better now. shall we take it like that, or shall i drop the latest commit?

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 2]
```diff
@@ -285,8 +285,10 @@ def restore_database
       end
 
       def proposal?(file)
-        !file.basename.to_s.match(/(_network\.json$)|(^template-(.*).json$)|(^queue\.json$)/) && \
-          file.to_s =~ /knife\/databags\/crowbar/
+        return false if file.basename.to_s =~ /^template-(.*).json$/
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: Can you add back the `file.to_s =~ /knife\/databags\/crowbar/` check? This is definitely a requirement. And then, with the check of the content of the data bag item, I think we're good.

+1 after that.

**Comment by MaximilianMeister** [comment]
> Can you add back the `file.to_s =~ /knife\/databags\/crowbar/` check? 

done

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 3]
```diff
@@ -285,7 +285,10 @@ def restore_database
       end
 
       def proposal?(file)
-        !file.basename.to_s.match(/(_network\.json$)|(^template-(.*).json$)|(^queue\.json$)/) && \
+        return false if file.basename.to_s =~ /^template-(.*).json$/
+        raw_data = JSON.parse(file.read)["raw_data"]
+        return false if raw_data.nil?
+        raw_data.key?("attributes") && raw_data.key?("deployment") && \
           file.to_s =~ /knife\/databags\/crowbar/
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: Heh, I would have added that check with the template one to avoid loading the file, but +1.

**Comment by MaximilianMeister** [comment]
> Heh, I would have added that check with the template one to avoid loading the file, but +1.

updated

**Comment by rhafer** [comment]
+1

**Comment by vuntz** [comment]
Still +1

**Pull request merged**